lib/gourami/coercer.rb

```diff
@@ -28,7 +28,7 @@ def setter_filter(attribute_name, value, options)
     #
     # @return [String, nil]
     def coerce_string(value, options = {})
-      return if value.nil? && options.fetch(:allow_nil, true)
+      return if (value.nil? || value.empty?) && options.fetch(:allow_nil, true)
```

Comment:
This is a change in behavior, it's pretty risky to change this without introducing it as either opt-in or opt-out.

This could cause not null constraints in our DB start failing right?

Reply:
E.g. opt-in:

```ruby
attribute(:spotify_id, :nil_when_empty => true)
attribute(:something_not_nullable)
```

Vs opt-out:

```ruby
attribute(:spotify_id)
attribute(:something_not_nullable, :nil_when_empty => false)
```